netket/graph/grid.py

```diff
@@ -185,6 +230,98 @@ def translations(self, dim: int = None, period: int = 1) -> SymmGroup:
 
         return SymmGroup([Identity()] + translations, graph=self)
 
+    def planar_rotation(self, axes: tuple = (0, 1), period: int = 1) -> SymmGroup:
+        """
+        Returns SymmGroup consisting of rotations about the origin in the plane defined by axes
+
+        Arguments:
+            axes: Axes that define the plane of rotation specified by dims.
+            period: Period of the rotations; should be a divisor of 4.
+        """
+
+        dims = tuple(self.length)
+
+        if not len(axes) == 2:
+            raise ValueError(f"Plane is specified by two axes")
+        if len(dims) < 2:
+            raise ValueError(f"Rotations not defined for 1d systems")
+        if _np.any(axes) > len(dims) - 1:
+            raise ValueError(f"Axis specified not in dims")
+
+        if self.length[axes[0]] == self.length[axes[1]]:
+            basis = (range(0, 4, period), [axes])
+        else:
+            basis = (range(0, 4, 2 * period), [axes])
+
+        rotations = itertools.product(*basis)
+        next(rotations)
+
+        rotations = [PlanarRotation(num, ax, dims) for (num, ax) in rotations]
+
+        return SymmGroup([Identity()] + rotations, graph=self)
+
+    def axis_reflection(self, axis: int = 0) -> List[List[int]]:
+        """
+        Returns SymmGroup consisting of identity and the lattice
+        reflected about the hyperplane axis = 0
+
+        Arguments:
+            axis: Axis to be reflected about
+        """
+
+        if abs(axis) > len(self.length) - 1:
+            raise ValueError(f"Axis specified not in dims")
+
+        dims = tuple(self.length)
+        basis = (range(0, 2), [axis])
+        reflections = itertools.product(*basis)
+        next(reflections)
+
+        reflections = [Reflection(num, ax, dims) for (num, ax) in reflections]
+
+        return SymmGroup([Identity()] + reflections, graph=self)
+
+    def rotations(self, period: int = 1) -> SymmGroup:
+        """
+        Returns all possible rotations of a hypercube lattice
+
+        The rotations are a subset of the permutations returned by
+        `self.automorphisms()`.
+
+        Arguments:
+            period: Period of the rotations; should be a divisor of 4.
+        """
+
+        axes = itertools.combinations(range(len(self.length)), 2)
+        group = SymmGroup([Identity()], graph=self)
+
+        for axs in axes:
+            group = group @ self.planar_rotation(axs, period)
+
+        return group
+
+    def space_group(self) -> SymmGroup:
+        """
+        Returns the full space grouup of a hypercube lattice
+
+        The space group is a subset of the permutations returned by
+        `self.automorphisms()`.
+
+        """
+
+        return self.rotations() @ self.axis_reflection()
+
+    def lattice_group(self) -> SymmGroup:
+        """
+        Returns the full lattice grouup of a hypercube lattice
```

Comment:
```suggestion
        Returns the full lattice symmetry group consisting of rotations, reflections, and periodic translation.
```
(I wouldn't say hypercube, as this works also for a rectangular shape now.)